pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -570,19 +571,12 @@ func (vs *VSphere) DiskIsAttached(volPath string, nodeName k8stypes.NodeName) (b
 }
 
 // DisksAreAttached returns if disks are attached to the VM using controllers supported by the plugin.
-func (vs *VSphere) DisksAreAttached(volPaths []string, nodeName k8stypes.NodeName) (map[string]bool, error) {
-	disksAreAttachedInternal := func(volPaths []string, nodeName k8stypes.NodeName) (map[string]bool, error) {
-		attached := make(map[string]bool)
-		if len(volPaths) == 0 {
+func (vs *VSphere) DisksAreAttached(nodeVolumes map[k8stypes.NodeName][]string) (map[k8stypes.NodeName]map[string]bool, error) {
```

Comment:
We just have to be careful and make sure we handle cases of paginating the results if single query can't return all the results. 

On AWS at least, I noticed we had to paginate the bulkverify request for correctly fetching the results.

Reply:
In case of vSphere, we will be able to retrieve all the results in one call.